app/api/business/validators/supplier_validator.py

```diff
@@ -170,72 +167,83 @@ def __validate_document(self, documents, name, now, has_expiry=True):
+                'message': 'You must set a filename for your {document_name} document.'
                 .format(
                     document_name=document_name
                 ),
                 'severity': 'warning',
-                'step': 'documents'
+                'step': 'documents',
+                'id': 'S008-{}'.format(name)
             })
 
         if has_expiry:
             expiry = document.get('expiry')
             if not expiry and document_required:
                 errors.append({
-                    'message': 'Your seller profile has no expiry date for the '
-                               '{document_name} document you uploaded.'
+                    'message': 'You must set an expiry date for your {document_name} document.'
                     .format(
                         document_name=document_name
                     ),
                     'severity': 'warning',
-                    'step': 'documents'
+                    'step': 'documents',
+                    'id': 'S009-{}'.format(name)
                 })
             elif document_required:
                 try:
                     expiry_date = pendulum.parse(expiry)
-                    if now > expiry_date.date():
+                    if now.date() > expiry_date.date():
+                        e = pendulum.instance(expiry_date)
+                        delta = now.diff(e).in_days()
+                        message = 'Your {document_name} document has expired. Please upload an updated version.'
+                        if delta > 28:
+                            message = (
+                                'Your {document_name} document has expired.  Please upload an updated version. '
+                                'Failure to provide this documentation may result in the '
+                                'suspension of your seller profile.'
+                            )
                         errors.append({
-                            'message': 'The {document_name} document you uploaded as part of your '
-                                       'seller profile has expired.'
+                            'message': message
                             .format(
                                 document_name=document_name
                             ),
                             'severity': 'warning',
-                            'step': 'documents'
+                            'step': 'documents',
+                            'id': 'S010-{}'.format(name)
                         })
-                    elif now.add(months=1) > expiry_date.date():
+                    elif now.add(days=28).date() > expiry_date.date():
                         errors.append({
-                            'message': 'The {document_name} document you uploaded as part of your '
-                                       'seller profile is about to expire.'
+                            'message': 'Your {document_name} document will expire in 28 days. '
```

Comment:
```suggestion
                            'message': 'Your {document_name} document will expire on {expiry_date}. '
```
Text provided by content.